### PR TITLE
Some updates.

### PR DESCRIPTION
**Pull request by ralfbiedert**
I was looking for a small, portable neural net library and found your project. 

Right now I'm only toying around and try to understand how things work. In that process I upgraded the project to Rust 2018 and applied various clippy and warning fixes, as well as `cargo fmt`.
 
I was also curious if you accept PRs, and if you consider resuming development (incl. publishing new releases).

### CODE REVIEW

**Comment by ralfbiedert** [comment]
Ah, and while I haven't come that far, I was wondering in what state 0.5 is in? My first impression was that it looks nice, but I'm not quite understanding the `topology` / `topology_v4` plans yet.

**Comment by Robbepop** [comment]
Hey, thank you for the PR.
It looks massive, but probably mainly due to the rustfmt changes.
Can you please separate the rustfmt changes into another PR so that reviewing the other more critical parts is actually feasible?

I ll gladly accept pull requests and merge them if they meet some quality criteria.
The v0.5 version never made it stable due to some design mistakes that demotivated me. But all in all it should be a major improvement compared to the current state already.
I don't think that v0.5 required a lot of further work for finalization. It is actually a pity that I dropped it.

**Comment by ralfbiedert** [comment]
I've redone most parts, this time without `fmt`. ~It would be nice if you could push a `rustfmt.toml` you prefer since it's hard for me to replicate the existing style manually~ (TIL it can also be `.rustfmt.toml` which I didn't see).

**Comment by ralfbiedert** [comment]
Strange that some unit tests fail, such as `before_and_after_elapsed` ... I'll have to investigate later.

**Comment by ralfbiedert** [comment]
Ok, based on #9 it looks like even without any changes in the code the appveyor CI tests started failing on their own in the meantime.

**Review by Robbepop — APPROVED** [review]
LGTM

We can merge if you want.

**Comment by ralfbiedert** [comment]
Also fixed the CI problems. Yes, please merge. 

Unless you disagree I want to PR a `cargo fmt` next ...

File: src/lib.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@
 //! 	.learn_momentum(0.6) // use the given learn momentum
 //! 	.log_config(LogConfig::Iterations(100)) // log state every 100 iterations
 //! 	.scheduling(Scheduling::Random)         // use random sample scheduling
-//! 	.criterion(Criterion::RecentMSE(0.03))  // train until the recent MSE is below 0.03
+//! 	.criterion(Criterion::RecentMSE(0.0001))  // train until the recent MSE is below 0.0001
```
Comment (Robbepop) [inline src/lib.rs:47]: was this tested so that the tests wont take ages?

Comment (ralfbiedert) [reply src/lib.rs:47]: Tested on my machine, and the CI. 

The alternative is switching this test off completely for now.

I don't think there is a good solution, apart from using a seeded random number generator. But I don't know how much work that would be.

**Comment by Robbepop** [comment]
Thank you for your work!

**Pull request merged**